snakemake_executor_plugin_kubernetes/__init__.py

```
@@ -223,55 +244,64 @@ def run_job(self, job: JobExecutorInterface):
+        import json
+        self.logger.debug("Pod specification:")
+        self.logger.debug(json.dumps(body.to_dict(), indent=2))
+
+        # Try creating the pod with exception handling
+        try:
+            pod = self._kubernetes_retry(
+                lambda: self.kubeapi.create_namespaced_pod(self.namespace, body)
+            )
+        except kubernetes.client.rest.ApiException as e:
+            self.logger.error(f"Failed to create pod: {e}")
+            raise WorkflowError(f"Failed to create pod: {e}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)